application/engine/services/context_budget_allocator.py

```
@@ -1987,64 +1987,26 @@ def _build_character_state_lock_block(self, novel_id: str, chapter_number: int)
 
         从本章 cast plan 读取 normal/minor 角色，避免 Bible 前 7 个角色污染上下文。
         """
+        # 单一路径：仅使用 character kernel（含统一投影）；失败显式记录并返回空，不回退旧 vector manager。
```

Ruff (0.16.2)

[warning] Comment contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF003)